data_write/data_write.py

```diff
@@ -600,23 +789,33 @@ def write_bfiq_params(parameters_holder):
                 parameters['antenna_arrays_order'] = []
 
                 flattened_data = []
+                num_antenna_arrays = 0
                 if "main" in bfiq[slice_id]:
+                    num_antenna_arrays += 1 
                     parameters['antenna_arrays_order'].append("main")
                     flattened_data.append(bfiq[slice_id]['main']['data'])
                 if "intf" in bfiq[slice_id]:
+                    num_antenna_arrays += 1 
                     parameters['antenna_arrays_order'].append("intf")
                     flattened_data.append(bfiq[slice_id]['intf']['data'])
 
                 flattened_data = np.concatenate(flattened_data)
                 parameters['data'] = flattened_data
 
-                parameters['num_samps'] = np.uint32(bfiq[slice_id].pop('num_samps', None))
-                parameters['data_dimensions'] = np.array([len(bfiq[slice_id].keys()),
+                parameters['num_samps'] = np.uint32(bfiq[slice_id]['num_samps'])
+                parameters['data_dimensions'] = np.array([num_antenna_arrays,
                                                           integration_meta.nave,
                                                           len(parameters['beam_nums']),
                                                           parameters['num_samps']], dtype=np.uint32)
 
 
+                for field in list(parameters.keys()):
+                    if field not in needed_fields:
+                        parameters.pop(field, None)
```

Comment:
Again just curious if you ever expect this to happen, that you'll have any non-needed fields.
EDIT:  I think I understand now that you'll have fields from bfiq not needed in raw for example, please let me konw if this is correct.

Reply:
yes